main.c

```diff
@@ -146,18 +212,17 @@ static void linux_boot(const struct boot_params boot_params)
     die_on(LINUX_HEADER_OFFSET + linux_header_size > sizeof(*linux_params),
            "Invalid linux header size");
 
-    memcpy((char *)linux_params + LINUX_HEADER_OFFSET,
-           (const char *)(boot_params.kernel_addr + LINUX_HEADER_OFFSET),
+    memcpy((uint8_t *)linux_params + LINUX_HEADER_OFFSET,
+           (const uint8_t *)(uintptr_t)(l_params->pref_address + LINUX_HEADER_OFFSET),
```

Comment:
otherwise there is a compiler error

Reply:
To avoid ` warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]`. In this case, it would be harmless, but there is some sadness regarding sign-extension that make some of these conversions highly unintuitive...